<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>963</width>
    <height>671</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>QDSP example: chirp compression &amp; threads</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <layout class="QGridLayout" name="gridLayout">
      <item row="0" column="3">
       <widget class="QLabel" name="label_2">
        <property name="text">
         <string>Frequency deviation</string>
        </property>
       </widget>
      </item>
      <item row="0" column="7">
       <layout class="QHBoxLayout" name="horizontalLayout_6">
        <item>
         <widget class="QSlider" name="pulseDelaySlider">
          <property name="maximum">
           <number>99</number>
          </property>
          <property name="singleStep">
           <number>1</number>
          </property>
          <property name="pageStep">
           <number>20</number>
          </property>
          <property name="value">
           <number>10</number>
          </property>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="periodCountLabel_2">
          <property name="text">
           <string>n/a</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item row="1" column="7">
       <widget class="QComboBox" name="methodComboBox">
        <property name="minimumSize">
         <size>
          <width>0</width>
          <height>0</height>
         </size>
        </property>
        <item>
         <property name="text">
          <string>Time domain</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Frequency domain</string>
         </property>
        </item>
       </widget>
      </item>
      <item row="0" column="1">
       <layout class="QHBoxLayout" name="horizontalLayout_4">
        <item>
         <widget class="QSlider" name="chirpSizeSlider">
          <property name="value">
           <number>0</number>
          </property>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="chirpSizeValueLabel">
          <property name="text">
           <string>n/a</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item row="1" column="3">
       <widget class="QLabel" name="label_3">
        <property name="text">
         <string>Window function</string>
        </property>
       </widget>
      </item>
      <item row="0" column="13">
       <layout class="QHBoxLayout" name="horizontalLayout_8">
        <item>
         <widget class="QSlider" name="periodCountSlider">
          <property name="maximum">
           <number>100</number>
          </property>
          <property name="singleStep">
           <number>10</number>
          </property>
          <property name="pageStep">
           <number>20</number>
          </property>
          <property name="value">
           <number>0</number>
          </property>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="periodCountLabel">
          <property name="text">
           <string>n/a</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item row="0" column="9">
       <widget class="QLabel" name="label_8">
        <property name="text">
         <string>Period size</string>
        </property>
       </widget>
      </item>
      <item row="1" column="11">
       <widget class="Line" name="line_8">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="label_6">
        <property name="text">
         <string>Noise dispersion</string>
        </property>
       </widget>
      </item>
      <item row="1" column="9">
       <widget class="QLabel" name="label_7">
        <property name="text">
         <string>Y-axis scale</string>
        </property>
       </widget>
      </item>
      <item row="1" column="8">
       <widget class="Line" name="line_6">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
       </widget>
      </item>
      <item row="0" column="6">
       <widget class="QLabel" name="label_10">
        <property name="text">
         <string>Pulse delay</string>
        </property>
       </widget>
      </item>
      <item row="0" column="12">
       <widget class="QLabel" name="label_9">
        <property name="text">
         <string>Period count</string>
        </property>
       </widget>
      </item>
      <item row="0" column="8">
       <widget class="Line" name="line_5">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
       </widget>
      </item>
      <item row="0" column="11">
       <widget class="Line" name="line_7">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
       </widget>
      </item>
      <item row="0" column="2">
       <widget class="Line" name="line_2">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
       </widget>
      </item>
      <item row="1" column="4">
       <widget class="QComboBox" name="windowComboBox">
        <property name="minimumSize">
         <size>
          <width>100</width>
          <height>0</height>
         </size>
        </property>
       </widget>
      </item>
      <item row="1" column="2">
       <widget class="Line" name="line">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
       </widget>
      </item>
      <item row="1" column="5">
       <widget class="Line" name="line_4">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
       </widget>
      </item>
      <item row="1" column="6">
       <widget class="QLabel" name="label_4">
        <property name="text">
         <string>Compression</string>
        </property>
       </widget>
      </item>
      <item row="1" column="10">
       <widget class="QComboBox" name="scaleComboBox">
        <property name="minimumSize">
         <size>
          <width>80</width>
          <height>0</height>
         </size>
        </property>
        <item>
         <property name="text">
          <string>Linear</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Logarithmic</string>
         </property>
        </item>
       </widget>
      </item>
      <item row="0" column="4">
       <layout class="QHBoxLayout" name="horizontalLayout_3">
        <item>
         <widget class="QSlider" name="deviationSlider">
          <property name="maximum">
           <number>100</number>
          </property>
          <property name="value">
           <number>0</number>
          </property>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="deviationValueLabel">
          <property name="text">
           <string>n/a</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="label_5">
          <property name="text">
           <string>%</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item row="1" column="1">
       <layout class="QHBoxLayout" name="horizontalLayout_5">
        <item>
         <widget class="QSlider" name="noiseDispersionSlider">
          <property name="maximum">
           <number>1000</number>
          </property>
          <property name="value">
           <number>0</number>
          </property>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="noiseDeviationLabel">
          <property name="text">
           <string>n/a</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item row="0" column="10">
       <layout class="QHBoxLayout" name="horizontalLayout_7">
        <item>
         <widget class="QSlider" name="periodSizeSlider">
          <property name="maximum">
           <number>100</number>
          </property>
          <property name="value">
           <number>0</number>
          </property>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="periodSizeLabel">
          <property name="text">
           <string>n/a</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item row="0" column="0">
       <widget class="QLabel" name="label">
        <property name="text">
         <string>Chirp size</string>
        </property>
       </widget>
      </item>
      <item row="0" column="5">
       <widget class="Line" name="line_3">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
       </widget>
      </item>
      <item row="1" column="12">
       <widget class="QPushButton" name="resetPushButton">
        <property name="text">
         <string>Reset controls</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <widget class="QCustomPlot" name="plot" native="true">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>963</width>
     <height>21</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <customwidgets>
  <customwidget>
   <class>QCustomPlot</class>
   <extends>QWidget</extends>
   <header>qcustomplot.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>chirpSizeSlider</sender>
   <signal>valueChanged(int)</signal>
   <receiver>chirpSizeValueLabel</receiver>
   <slot>setNum(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>165</x>
     <y>53</y>
    </hint>
    <hint type="destinationlabel">
     <x>228</x>
     <y>53</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>deviationSlider</sender>
   <signal>valueChanged(int)</signal>
   <receiver>deviationValueLabel</receiver>
   <slot>setNum(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>360</x>
     <y>53</y>
    </hint>
    <hint type="destinationlabel">
     <x>462</x>
     <y>53</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>periodCountSlider</sender>
   <signal>valueChanged(int)</signal>
   <receiver>periodCountLabel</receiver>
   <slot>setNum(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>567</x>
     <y>115</y>
    </hint>
    <hint type="destinationlabel">
     <x>579</x>
     <y>115</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pulseDelaySlider</sender>
   <signal>valueChanged(int)</signal>
   <receiver>periodCountLabel_2</receiver>
   <slot>setNum(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>604</x>
     <y>47</y>
    </hint>
    <hint type="destinationlabel">
     <x>704</x>
     <y>43</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
